google/cloud/pubsub/internal/streaming_subscription_batch_source.cc

```diff
@@ -236,16 +236,22 @@ void StreamingSubscriptionBatchSource::ExtendLeases(
   }
   lk.unlock();
   for (auto& r : split) {
+    lk.lock();
+    auto modack_request_id = modack_request_id_++;
+    lk.unlock();
+    callback_->StartModackSpan(request, modack_request_id);
     (void)stub_
         ->AsyncModifyAckDeadline(cq_, std::make_shared<grpc::ClientContext>(),
                                  options_, r)
-        .then([cb = callback_, r](auto f) {
-          auto result = f.get();
-          for (auto const& ack_id : r.ack_ids()) {
-            cb->ModackEnd(ack_id);
-          }
-          return result;
-        });
+        .then(
+            [cb = callback_, modack_request_id = modack_request_id, r](auto f) {
+              auto result = f.get();
+              cb->EndModackSpan(modack_request_id);
+              for (auto const& ack_id : r.ack_ids()) {
+                cb->ModackEnd(ack_id);
```

Comment:
Is there a preferred ordering to the `EndModackSpan()` and the `ModackEnd()`?

Reply:
I don't think it makes a big difference either way. I moved it though because it might make more sense logically to have it mark the event, then end the span.

Reply:
Not really, I moved the event creation before the span end.